src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  const result = {};
+
+  const rules = sourceString
+    .split(';')
+    .map((rule) => rule.trim())
+    .filter(Boolean);
+
+  rules.forEach((rule) => {
+    const [key, ...valueParts] = rule.split(':');
```

Comment:
The use of destructuring with `const [key, ...valueParts] = rule.split(':');` is a good approach to separate the key and value parts. However, ensure that the input string is well-formed, as this code assumes that there is at least one colon in each rule.